nsv13/code/modules/overmap/weapons/mines.dm

```diff
@@ -0,0 +1,99 @@
+	var/damage_flag = "overmap_heavy"
+	alpha = 50 //They're supposed to be sneaky, their main advantage is being cloaked
+
+/obj/structure/space_mine/Initialize(mapload, var/list/coordinates, var/new_faction, var/datum/star_system/system)
+	. = ..()
+	if(system)
+		current_system = system
+	else //Someone is probably spawning us on the overmap, so we assume it's next to the main ship
+		current_system = SSstar_system.find_main_overmap().current_system
+	current_system.system_contents |= src
+	if(new_faction)
+		faction = new_faction
+	update_icon()
+	if(coordinates)
+		x = coordinates["x"]
+		y = coordinates["y"]
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/structure/space_mine/Destroy(force)
+	current_system?.contents_positions.Remove(src)
+	current_system?.system_contents.Remove(src)
+	RemoveElement(/datum/element/connect_loc)
+	. = ..()
+
+/// This makes us not drift like normal objects in space do
+/obj/structure/space_mine/Process_Spacemove(movement_dir = 0)
+	return 1
+
+/obj/structure/space_mine/proc/on_entered(datum/source, atom/movable/AM)
+	SIGNAL_HANDLER
+
+	if(!(istype(AM, /obj/structure/overmap) || istype(AM, /obj/item/projectile)))
+		return
+
+	switch(AM.type)
+		if(/obj/item/projectile)
+			var/obj/item/projectile/P = AM
+			if(P.faction != faction || !(P.faction == "nanotrasen" || P.faction == "solgov") && (faction == "nanotrasen" || faction == "solgov"))
+				P.Impact(src)
+
+		if(/obj/structure/overmap)
+			var/obj/structure/overmap/OM = AM
+			if(OM.faction != faction || !(OM.faction == "nanotrasen" || OM.faction == "solgov") && (faction == "nanotrasen" || faction == "solgov"))
+				mine_explode(OM)
+
+/obj/structure/space_mine/update_icon(updates)
+	. = ..()
+	icon_state = "mine_[faction]"
+
+/obj/structure/space_mine/obj_break(damage_flag)
+	if(prob(80))
+		obj_destruction()
+	else //Whoops, IFF broke!
+		faction = "unaligned"
+
+/obj/structure/space_mine/obj_destruction(damage_flag)
+	mine_explode() //Why you mine explode? To the woods with you
+	..(damage_flag)
+
+/obj/structure/space_mine/proc/mine_explode(obj/structure/overmap/OM)
+	var/armour_penetration
+	if(OM) //You just flew into a mine
+		armour_penetration = 20 //It's blowing up right next to you, this is what it was designed for
+		if(OM.use_armour_quadrants)
+			OM.take_quadrant_hit(OM.run_obj_armor(damage, damage_type, damage_flag, null, armour_penetration), OM.quadrant_impact(src))
+		else
+			OM.take_damage(damage, damage_type, damage_flag, FALSE, TRUE)
+		if(OM.linked_areas) //Hope nothing precious was in that room.
+			var/area/A = pick(OM.linked_areas)
+			var/turf/T = pick(get_area_turfs(A))
+			new /obj/effect/temp_visual/explosion_telegraph(T, damage)
+	else
+		for(var/obj/structure/overmap/O in orange(2)) //You're in range! Keep in mind this affects *all* ships, explosions don't discriminate between friend and foe
+			OM = O
```

Comment:
Should be called even if OM exists for the "fun" of it?
Maybe 
```
if(OM) listvar =  orange(2) - OM 
else listvar = orange(2)
```

Reply:
orange, so by my logic I excluded whatever hit it, but I seem to have forgotten our ships are often not very neatly positioned.

Reply:
WAIT orange needs the center as second arg though right? I forgot that completely!

Reply:
Oh also, this only happens when there's no direct detonation in the first place (no OM var) because of the if check. When a ship detonates the mine it absorbs the full blast, but otherwise the detonation is applied to ships around it.

Reply:
Yeah I was suggesting running it always because kaboom is kaboom, but I guess it absorbing most of the hit makes sense.